pkg/controller/mcmhub/mcmhub_controller.go

```diff
@@ -628,9 +628,12 @@ func (r *ReconcileSubscription) IsSubscriptionCompleted(subKey types.NamespacedN
 	}
 
 	// need to wait for managed cluster reporting back
+	// fix: When placmentrule doesn't have target cluster decision list, managed clusters status is empty.
+	// In this case, main subscription does nothing, It should be regarded as complete sub,
+	// so that posthook will be able to execute
 	managedStatus := subIns.Status.Statuses
 	if len(managedStatus) == 0 {
-		return false, nil
+		return true, nil
```

Comment:
@xiangjingli 

hmmm, I was thinking post hook is triggered after all the managed cluster is done.
If we allow the empty status, then right after the main reconcile, the post hook will be executed...

to address the new case, we might want to check if the decision list is empty or not, if it’s empty, we can return true right aways, if it’s not, then we will have to wait for the managed cluster to report back